app/rtc_s1.py

```diff
@@ -0,0 +1,1262 @@
+       full_log_formatting : bool
+              Flag to enable full formatting of logged messages
+
+       Returns
+       -------
+       logger : logging.Logger
+              Logger object
+    """
+    # create logger
+    logger.setLevel(logging.DEBUG)
+
+    # create console handler and set level to debug
+    ch = logging.StreamHandler()
+    ch.setLevel(logging.DEBUG)
+
+    # create formatter
+    # configure full log format, if enabled
+    if full_log_formatting:
+        msgfmt = ('%(asctime)s.%(msecs)03d, %(levelname)s, DSWx-HLS, '
+                  '%(module)s, 999999, %(pathname)s:%(lineno)d, "%(message)s"')
+
+        formatter = logging.Formatter(msgfmt, "%Y-%m-%d %H:%M:%S")
+    else:
+        formatter = logging.Formatter('%(message)s')
+
+    # add formatter to ch
+    ch.setFormatter(formatter)
+
+    # add ch to logger
+    logger.addHandler(ch)
+
+    if log_file:
+        file_handler = logging.FileHandler(log_file)
+
+        file_handler.setFormatter(formatter)
+
+        # add file handler to logger
+        logger.addHandler(file_handler)
+
+    sys.stdout = Logger(logger, logging.INFO)
+    sys.stderr = Logger(logger, logging.ERROR, prefix='[StdErr] ')
+
+    return logger
+
+
+def _update_mosaic_boundaries(mosaic_geogrid_dict, geogrid):
+    xf = geogrid.start_x + geogrid.spacing_x * geogrid.width
+    yf = geogrid.start_y + geogrid.spacing_y * geogrid.length
+    if ('x0' not in mosaic_geogrid_dict.keys() or
+            geogrid.start_x < mosaic_geogrid_dict['x0']):
+        mosaic_geogrid_dict['x0'] = geogrid.start_x
+    if ('xf' not in mosaic_geogrid_dict.keys() or
+            xf > mosaic_geogrid_dict['xf']):
+        mosaic_geogrid_dict['xf'] = xf
+    if ('y0' not in mosaic_geogrid_dict.keys() or
+            geogrid.start_y > mosaic_geogrid_dict['y0']):
+        mosaic_geogrid_dict['y0'] = geogrid.start_y
+    if ('yf' not in mosaic_geogrid_dict.keys() or
+            yf < mosaic_geogrid_dict['yf']):
+        mosaic_geogrid_dict['yf'] = yf
+    if 'dx' not in mosaic_geogrid_dict.keys():
+        mosaic_geogrid_dict['dx'] = geogrid.spacing_x
+    else:
+        assert(mosaic_geogrid_dict['dx'] == geogrid.spacing_x)
+    if 'dy' not in mosaic_geogrid_dict.keys():
+        mosaic_geogrid_dict['dy'] = geogrid.spacing_y
+    else:
+        assert(mosaic_geogrid_dict['dy'] == geogrid.spacing_y)
+    if 'epsg' not in mosaic_geogrid_dict.keys():
+        mosaic_geogrid_dict['epsg'] = geogrid.epsg
+    else:
+        assert(mosaic_geogrid_dict['epsg'] == geogrid.epsg)
+
+
+def _get_raster(output_dir, ds_name, ds_hdf5, dtype, shape,
+                radar_grid_file_dict, output_obj_list, flag_save_vector_1,
+                extension):
+    if flag_save_vector_1 is not True:
+        return None
+
+    output_file = os.path.join(output_dir, ds_name) + '.' + extension
+    raster_obj = isce3.io.Raster(
+        output_file,
+        shape[2],
+        shape[1],
+        shape[0],
+        dtype,
+        "GTiff")
```

Comment:
Could this be generalized to "extension"

Reply:
Uhm, this is the file format of the temporary files. For example, if the output format is set to `NETCDF`, the extension is `nc`, and the file format for temporary files will still be `GTiff`. Is that OK?